app/services/umrdr/date_coverage_service.rb

```diff
@@ -0,0 +1,130 @@
+        reverse_interval!(interv) if interval_reversed?(interv) 
+        make_begin_params(interv.begin).merge(make_end_params(interv.end))
+      end
+
+      private
+
+      # Determines if an intervals endings are reverse.  Specifically, where the begin 
+      # date is later that the end date.  An interval with reversed starting and ending
+      # dates will have from and to attributes, but begin and end will be nil
+      # == Parameters:
+      # interv::
+      #   [EDTF::Interval] The interval to be tested. 
+      # == Returns:
+      #   [Boolean] True if the interval boundaries were reversed
+      def interval_reversed?(interv)
+        return false if interv.from == EDTF_OPEN_BEGIN || interv.to == EDTF_OPEN_END 
+        (interv.from && !interv.begin) && (interv.to && !interv.end)
+      end
+
+      # Reverse the begin and end of an interval.  This will bork a well formed interval.
+      # == Parameters:
+      # interv::
+      #   [EDTF::Interval] The interval to be reversed. 
+      # == Returns:
+      #   [EDTF::Interval] Interval with from and to in the reverse order from the input interval.
+      def reverse_interval!(interv)
+        tmp = interv.to
+        interv.to = interv.from
+        interv.from = tmp
+      end
+
+      def make_begin_params(date)
+        hsh = date_to_params date
+        {date_coverage_begin_year:  hsh[:year],
+         date_coverage_begin_month: hsh[:month],
+         date_coverage_begin_day:   hsh[:day]}
+      end
+
+      def make_end_params( date )
+        hsh = date_to_params date
+        {date_coverage_end_year:  hsh[:year],
+         date_coverage_end_month: hsh[:month],
+         date_coverage_end_day:   hsh[:day]}
+      end
+
+      # Given edtf date
+      # return hash with possible keys :year, :month, :day
+      def date_to_params( date )
+        return {} unless date
+        hsh = case date.precision
+        when :year
+          {year: date.year}
+        when :month
+          {year: date.year, month: date.month}
+        when :day
+          {year: date.year, month: date.month, day: date.day}
+        end
+        hsh[:year] = year_to_s hsh[:year]
+        hsh.each{|k,v| hsh[k] = v.to_s}
+      end
+
+      def year_to_s(val)
+        if val.negative?
+          sprintf('-%04d', val.abs)
+        else
+          sprintf('%04d', val)
+        end
+      end
+
+      # Given input
+      # Return the corresponding integer or nil
+      def integerize(input)
+        Integer(input,10) rescue nil
+      end
+
+      # Until 8601-2 gets finalized, edtf ranges do not accept :open for begin.
+      def make_begin_date(y = nil, m = nil, d = nil)
+        make_date(y,m,d) || EDTF_OPEN_BEGIN
+      end
+
+      def make_end_date(y = nil, m = nil, d = nil)
+        make_date(y,m,d) || EDTF_OPEN_END
+      end
+
+      # Takes year, month, and day integers.
+      # Returns edtf date or nil
+      def make_date(y = nil, m = nil, d = nil)
+        dt = case
```

Comment:
If people can do something like a year and a day I would use named arguments instead of positional arguments to make it clear what is going on and to prevent odd bugs.

> On May 19, 2016, at 11:49 AM, Colin <notifications@github.com> wrote:
> 
> In app/services/umrdr/date_coverage_service.rb:
> 
> > +        Integer(input,10) rescue nil
> > +      end
> > +
> > +      # Until 8601-2 gets finalized, edtf ranges do not accept :open for begin.
> > +      def make_begin_date(y = nil, m = nil, d = nil)
> > +        make_date(y,m,d) || EDTF_OPEN_BEGIN
> > +      end
> > +
> > +      def make_end_date(y = nil, m = nil, d = nil)
> > +        make_date(y,m,d) || EDTF_OPEN_END
> > +      end
> > +
> > +      # Takes year, month, and day integers.
> > +      # Returns edtf date or nil
> > +      def make_date(y = nil, m = nil, d = nil)
> > +        dt = case
> Could do decorator. Array based approaches aren't garunteed to work when only given a month and a day, or a year and a day. Bad array lengths happen.
> 
> A decorator around Date is a neat idea.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>